app/models/supported_ssg.rb

```diff
@@ -0,0 +1,47 @@
+# frozen_string_literal: true
+
+require 'yaml'
+
+# rubocop:disable Metrics/BlockLength
+SupportedSsg = Struct.new(:id, :package, :version, :profiles,
+                          :os_major_version, :os_minor_version,
+                          keyword_init: true) do
+  self::SUPPORTED_FILE = Rails.root.join('config/supported_ssg.yaml')
+
+  class << self
+    def all
+      raw_supported['supported'].map do |rhel, values|
+        major, minor = os_version(rhel)
+
+        new(
+          id: rhel,
+          os_major_version: major,
+          os_minor_version: minor,
+          **map_attributes(values)
+        )
+      end
+    end
+
+    def revision
+      raw_supported['revision']
+    end
+
+    private
+
+    def map_attributes(values)
+      attrs = values.slice(*members.map(&:to_s))
+      attrs['profiles'] = attrs['profiles']&.keys
```

Comment:
We will need to be able to get at `old_names` as well; probably not in the API, but certainly in other backend code. This logic to remove certain information from the config to serve in the API can happen in the serializer directly.

Reply:
Should I add it to the API as well, or a new (non-exposed) property would be fine?

Reply:
I don't think it's necessary for the API.

Reply:
Kept the profiles structure of YAML in the model. Moved transformation to serializer.